addons/web/static/src/views/fields/properties/properties_field.js

```diff
@@ -63,23 +63,49 @@ export class PropertiesField extends Component {
         this.definitionRecordField = field.definition_record;
 
         this.state = useState({
-            canChangeDefinition: true,
+            canChangeDefinition: false,
+            isEditMode: false,
             movedPropertyName: null,
-            showAddButton: this.props.showAddButton,
             unfoldedSeparators: this._getUnfoldedSeparators(),
         });
 
         // Properties can be added from the cogmenu of the form controller
         if (this.env.config?.viewType === "form") {
-            useBus(this.env.model.bus, "PROPERTY_FIELD:ADD_PROPERTY_VALUE", () => {
-                this.onPropertyCreate();
+            useBus(this.env.model.bus, "PROPERTY_FIELD:EDIT", () => {
+                if (!this.props.readonly && this.state.canChangeDefinition) {
+                    this.state.isEditMode = true;
+                    if (this.propertiesList.length === 0) {
+                        this.onPropertyCreate();
+                    }
+                }
             });
         }
 
         onWillStart(async () => {
-            await this._checkDefinitionAccess();
+            const canChangeDefinition = await this.checkDefinitionWriteAccess();
+            this.state.canChangeDefinition = !!canChangeDefinition;
+        });
+
+        /** @param {function} callback */
+        const whenRecordStoringPropertyDefinitionChanges = (callback) => {
```

Comment:
Do you need to make this a function? Can't you directly put the content of this function in the useEffect to simplify?

Reply:
Yes, I can simplify it.

I introduced the intermediate function to have a code that is self-descriptive: When we read the function named, we can have an idea of what the function does without looking at the implementation (here, we have: "when the record storing the property definition changes, then the given callback function is executed").

I will remove that intermediate function and add a small comment explaining what this `useEffect` does 😄